src/Npgsql/ConnectorPool.cs

```diff
@@ -618,19 +565,15 @@ void CheckInvariants(PoolState state)
         {
             if (state.Total > _max)
                 throw new NpgsqlException($"Pool is over capacity (Total={state.Total}, Max={_max})");
-            if (state.Waiting > 0 && state.Idle > 0)
-                throw new NpgsqlException($"Can't have waiters ({state.Waiting}) while there are idle connections ({state.Idle}");
             if (state.Idle < 0)
                 throw new NpgsqlException("Idle is negative");
             if (state.Busy < 0)
                 throw new NpgsqlException("Busy is negative");
-            if (state.Waiting < 0)
-                throw new NpgsqlException("Waiting is negative");
         }
 
         public void Dispose() => _pruningTimer?.Dispose();
 
-        public override string ToString() => State.ToString();
+        public override string ToString() => State.Copy().ToString();
```

Comment:
FYI `PoolState.ToString()` already calls `Copy()` internally.

Reply:
I'll remove it from the inner to string as it shouldn't concern itself with threading issues